In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_excel('data.xlsx', sheet_name='Data')

In [3]:
df = df.drop(['KV', 'thunhap', 'chitieu', 'songuoi'],1)

In [4]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

In [5]:
y = df.pop('Chi tiêu')

In [6]:
df['Khu vực'] = df['Khu vực'].replace('Nông thôn', 0)
df['Khu vực'] = df['Khu vực'].replace('Thành thị', 1)
df['Khu vực'] = df['Khu vực'].astype(int)
df['Khu vực'].head()

0    1
1    1
2    1
3    1
4    1
Name: Khu vực, dtype: int32

In [7]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

df = df.select_dtypes(exclude = 'object')
# VIF dataframe
vif_data = pd.DataFrame()
vif_data["feature"] = df.columns
  
# calculating VIF for each feature
vif_data["VIF"] = [variance_inflation_factor(df.values, i)
                          for i in range(len(df.columns))]
  
print(vif_data)

       feature       VIF
0      Khu vực  2.392896
1     Số người  3.985062
2     Thu nhập  6.754451
3  Chi ăn uống  9.077163


In [8]:
df = df.drop(['Chi ăn uống'],1)

In [9]:

results = sm.OLS(y, sm.add_constant(df)).fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:               Chi tiêu   R-squared:                       0.730
Model:                            OLS   Adj. R-squared:                  0.728
Method:                 Least Squares   F-statistic:                     374.0
Date:                Fri, 21 May 2021   Prob (F-statistic):          1.05e-117
Time:                        15:00:52   Log-Likelihood:                -2156.8
No. Observations:                 420   AIC:                             4322.
Df Residuals:                     416   BIC:                             4338.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          2.3987      5.882      0.408      0.6

</br> 

Kết luận:

- Biến `Khu vực` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.000)
- Biến `Số người` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.013)
- Biến `Thu nhập` có ý nghĩa đối với mô hình về mặt thống kê (với mức ý nghĩa (***) hay p-value = 0.000)


- Mô hình có thể giải thích được 72.8% sự thay đổi của biến `charges`
